paddle/framework/tensor.h

```diff
@@ -164,6 +170,80 @@ class Tensor {
   size_t offset_;
 };
 
+/*
+ * LODTensor
+ */
+class LODTensor {
+ public:
+// level_t save offsets of each unit.
+#ifdef PADDLE_ONLY_CPU
+  using level_t = std::vector<uint32_t>;
+#else
+  using level_t = thrust::device_vector<uint32_t>;
+#endif
+  // LOD stores offsets of each level of units, the largest units level first,
+  // then the smaller units level. Each level_t stores the offsets of units in
+  // Tesor.
+  typedef std::vector<level_t> lod_t;
+
+  explicit LODTensor() {}
+
+  LODTensor(LODTensor&& other)
+      : lod_start_pos_(std::move(other.lod_start_pos_)),
+        tensor_(other.tensor_) {}
+
+  /*
+   * Number of LODTensor's levels, each level has units of data, for example,
+   * in the sentence's view, article, paragraph, sentence are 3 levels.
+   */
+  size_t Levels() const {
+    return lod_start_pos_.get() ? lod_start_pos_->size() : 0UL;
+  }
+  /*
+   * Number of elements in a level.
+   */
+  size_t Elements(uint32_t level = 0) const {
```

Comment:
Please use `size_t` not `uint32_t` for index.

Please always use `size_t` for index type because it is used widely in C++ standard library.